editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -0,0 +1,378 @@
+
+        if (event.isStillSincePress() && event.getButton() == MouseButton.PRIMARY) {
+          // reset the UI
+          featureLayer.clearSelection();
+          editFeatureVBox.setDisable(true);
+
+          // select the clicked feature
+          selectFeature(point);
+        }
+
+        if (event.isStillSincePress() && event.getButton() == MouseButton.SECONDARY) {
+          // if a feature is selected and the current version is not the default, update the feature's geometry
+          if (selectedFeature != null && !serviceGeodatabase.getVersionName().equals(defaultVersionName)) {
+            Point mapPoint = mapView.screenToLocation(point);
+            selectedFeature.setGeometry(mapPoint);
+            updateFeature(selectedFeature);
+          }
+        }
+      });
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Create a new branch version using user defined values.
+   */
+  @FXML
+  private void handleCreateVersionButtonClicked() {
+
+    // validate version name input
+    String inputName = nameTextField.getText();
+   if (!isTextInputValid(inputName)) {
+     return;
+   };
+
+    // validate version access input
+    if (accessTypeComboBox.getSelectionModel().getSelectedItem() == null) {
+      showAlert("Please select an access level");
+      return;
+    }
+
+    // set the user defined name, access level and description as service version parameters
+    ServiceVersionParameters newVersionParameters = new ServiceVersionParameters();
+    newVersionParameters.setName(inputName);
+    newVersionParameters.setAccess(accessTypeComboBox.getSelectionModel().getSelectedItem());
+    newVersionParameters.setDescription(descriptionTextField.getText());
+
+    // update the UI
+    createVersionButton.setText("Creating version....");
+    createVersionButton.setDisable(true);
+
+    // create a new version with the specified parameters
+    ListenableFuture<ServiceVersionInfo> newVersion = serviceGeodatabase.createVersionAsync(newVersionParameters);
+    newVersion.addDoneListener(() -> {
+      try {
+
+        // get the name of the created version and switch to it
+        ServiceVersionInfo createdVersionInfo = newVersion.get();
+        userCreatedVersionName = createdVersionInfo.getName();
+        switchVersion(userCreatedVersionName);
+
+        // hide the form from the UI as the sample only allows 1 version to be created
+        createVersionVBox.setVisible(false);
+        switchVersionButton.setDisable(false);
+      } catch (Exception ex) {
+        // if there is an error creating a new version, display an alert and reset the UI
+        if (ex.getCause().toString().contains("The version already exists")) {
+          showAlert("A version with this name already exists.\\nPlease enter a unique name");
+        } else {
+          showAlert("Error creating new version");
+        }
```

Comment:
could opt for a one liner here too
```suggestion
 } else showAlert("Error creating new version");
```